### DISCUSSION TITLE
Ytfzf 2.0

### DISCUSSION THREAD

**Discussion by Euro20179** [pystardust/ytfzf#436] [General]
We didn't like the old code base so we decided to rewrite it.
The goal is to keep the codebase simple, and avoid features that lots of complexity.

---

To update to the new release
```sh
git clone https://github.com/pystardust/ytfzf
cd ytfzf
sudo make install man
```
If you do not wish to install man pages (highly unrecommended) run `sudo make install` instead.

---

 # Changes

**Assume your current config file doesn't work with this update, see ytfzf(1) and ytfzf(5)**.
Feel free to ask here or on the discord for help.

## Breaking Changes
* Many variables have been renamed
* Many config options (such as video_player) that were strings are now functions

## Additions
* Scrapers for invidious
* -I to get various information about a video instead of playing it
* support for w3m image display (although it's a bit buggy)
* submenus, eg: when clicking on a playlist, it scrapes the playlist and opens it
* multi-scrape, scrape multiple websites at once
    * `ytfzf -cO,Y search`
    * the above will scrape odysee and youtube.
* framework for playlists
    * To see what a playlist file looks like, check out the `~/.cache/ytfzf/watch_hist` file (after you watch a video and if you have history enabled)

* Modularity
    * ability to add things such as scrapers, and thumbnail viewers yourself

## Documentation
* All documentation has been moved into the man pages, instead of conf.sh
    * this allows us to change default settings to fix bugs as everyone seemed to copy and paste the default config file

## Removed
* Search history
    * however it is possible to implement this yourself in conf.sh using the on_search function

## Bugs
* At least 1 bug has been fixed
* Many bugs have probably been introduced and uncaught, please file a bug report if you find any :)

<hr /><em>This discussion was created from the release <a href='https://github.com/pystardust/ytfzf/releases/tag/'>Ytfzf 2.0</a>.</em>

**Comment by timsofteng** [comment]
Do you consider implementation of preview in external program? Like imv or whatever.
It might be useful for tiling window managers. In addition it's simple way to see preview on Wayland.

**Reply by Euro20179** [reply]
I implemented imv viewer, I should've kept the branch, but didn't, I'll probably do it again, cause it was actually pretty decent.
However currently there is kitty.

Edit:
to better answer the quesiton, I think it's a good idea to add alternatives for ueberzug, until we can find (or maybe create) something that replaces it, and that works on wayland and x11.
